src/Integration/Exclusions/AnalyzableFileIndicator.cs

```diff
@@ -27,9 +29,49 @@ namespace SonarLint.VisualStudio.Integration.Exclusions
     [PartCreationPolicy(CreationPolicy.Shared)]
     internal class AnalyzableFileIndicator : IAnalyzableFileIndicator
     {
+        private readonly IExclusionSettingsFileStorage exclusionSettingsFileStorage;
+        private readonly IGlobPatternMatcher globPatternMatcher;
+
+        [ImportingConstructor]
+        public AnalyzableFileIndicator(IExclusionSettingsFileStorage exclusionSettingsFileStorage)
+            : this(exclusionSettingsFileStorage, new GlobPatternMatcher())
+        {
+        }
+
+        internal AnalyzableFileIndicator(IExclusionSettingsFileStorage exclusionSettingsFileStorage, 
+            IGlobPatternMatcher globPatternMatcher)
+        {
+            this.exclusionSettingsFileStorage = exclusionSettingsFileStorage;
+            this.globPatternMatcher = globPatternMatcher;
+        }
+
         public bool ShouldAnalyze(string filePath)
         {
-            return true;
+            var serverExclusions = exclusionSettingsFileStorage.GetSettings();
+
+            if (serverExclusions == null)
+            {
+                return true;
+            }
+
+            var shouldAnalyze = IsIncluded() && !IsExcludedInProjectOrGlobally();
+
+            return shouldAnalyze;
+
+            bool IsIncluded() =>
+                serverExclusions.Inclusions == null ||
+                serverExclusions.Inclusions.Length == 0 ||
+                serverExclusions.Inclusions.Any(IsMatch);
+
+            bool IsExcludedInProjectOrGlobally() =>
```

Comment:
Do the nested functions improve the readability?

Reply:
You tell me :D